src/pages/pool/_id.vue

```diff
@@ -210,106 +127,120 @@ interface PoolPageData {
+        poolQuery.isIdle.value ||
+        Boolean(poolQuery.error.value)
+    );
+    const loadingPool = computed(() => poolQueryLoading.value || !pool.value);
+
     const {
       isStableLikePool,
       isLiquidityBootstrappingPool,
       isStablePhantomPool
     } = usePool(poolQuery.data);
-
-    const noInitLiquidity = computed(
-      () =>
-        !loadingPool.value &&
-        pool.value &&
-        Number(pool.value?.onchain?.totalSupply || '0') === 0
-    );
-
-    const communityManagedFees = computed(
-      () => pool.value?.owner == POOLS.DelegateOwner
+    //#endregion
+
+    //#region pool snapshot query
+    const poolSnapshotsQuery = usePoolSnapshotsQuery(
+      route.params.id as string,
+      undefined,
+      // in order to prevent multiple coingecko requests
+      { refetchOnWindowFocus: false }
     );
-    const feesManagedByGauntlet = computed(
-      () =>
-        communityManagedFees.value &&
-        POOLS.DynamicFees.Gauntlet.includes(data.id)
-    );
-    const feesFixed = computed(() => pool.value?.owner == POOLS.ZeroAddress);
-    const swapFeeToolTip = computed(() => {
-      if (feesManagedByGauntlet.value) {
-        return t('feesManagedByGauntlet');
-      } else if (communityManagedFees.value) {
-        return t('delegateFeesTooltip');
-      } else if (feesFixed.value) {
-        return t('fixedFeesTooltip');
-      } else {
-        return t('ownerFeesTooltip');
-      }
-    });
-
-    const poolQueryLoading = computed(
+    const isLoadingSnapshots = computed(
       () =>
-        poolQuery.isLoading.value ||
-        poolQuery.isIdle.value ||
-        poolQuery.error.value
+        poolSnapshotsQuery.isLoading.value || poolSnapshotsQuery.isIdle.value
     );
 
-    const loadingPool = computed(() => poolQueryLoading.value || !pool.value);
-
     const snapshots = computed(() => poolSnapshotsQuery.data.value?.snapshots);
     const historicalPrices = computed(
       () => poolSnapshotsQuery.data.value?.prices
     );
-    const isLoadingSnapshots = computed(
+    //#endregion
+
+    //#region APR query
+    const aprQuery = usePoolAprQuery(route.params.id as string);
+    const loadingApr = computed(
       () =>
-        poolSnapshotsQuery.isLoading.value || poolSnapshotsQuery.isIdle.value
+        aprQuery.isLoading.value ||
+        aprQuery.isIdle.value ||
+        Boolean(aprQuery.error.value)
     );
+    const poolApr = computed(() => aprQuery.data.value);
+    //#endregion
+
+    //#region Intersection Observer
+    const intersectionSentinel = ref<HTMLDivElement | null>(null);
+    const isSentinelIntersected = ref(false);
+    let observer: IntersectionObserver | undefined;
+
+    function addIntersectionObserver(): void {
+      if (
+        !('IntersectionObserver' in window) ||
+        !('IntersectionObserverEntry' in window) ||
+        !intersectionSentinel.value
+      ) {
+        isSentinelIntersected.value = true;
+        return;
+      }
```

Comment:
Where is this pattern coming from? I'd like to understand it better.

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API
in this case Intersection Observer helps to render Investments and Trades tables on scrolling to them. it increases the speed of first page render.